code/modules/mechanics/MechanicMadness.dm

```diff
@@ -3267,6 +3268,39 @@ ADMIN_INTERACT_PROCS(/obj/item/mechanics/trigger/button, proc/press)
 			tooltip_rebuild = 1
 			if (autoEval)
 				src.evaluate()
+	proc/compSetMode(var/datum/mechanicsMessage/input)
+		LIGHT_UP_HOUSING
+		tooltip_rebuild = 1
+		switch(input.signal)
+			if("add")
+				mode = "add"
+			if("sub")
+				mode = "sub"
+			if("div")
+				mode = "div"
+			if("mul")
+				mode = "mul"
+			if("mod")
+				mode = "mod"
+			if("pow")
+				mode = "pow"
+			if("rng")
+				mode = "rng"
+			if("gt")
+				mode = "gt"
+			if("lt")
+				mode = "lt"
+			if("gte")
+				mode = "gte"
+			if("lte")
+				mode = "lte"
+			if("eq")
+				mode = "eq"
+			if("neq")
+				mode = "neq"
+			else
+				return
```

Comment:
What's the point of this switch? If it's for validating the mode string then a `if (mode in list(...` would be a lot cleaner.

Reply:
There we go I think I did what you said, at least that looks tidier to me